### PR TITLE
Adding so many types

### PR DESCRIPTION
**Pull request by joedevivo**

### CODE REVIEW

File: src/webmachine.erl [hunk 1]
```diff
@@ -19,6 +19,17 @@
 -author('Andy Gross <andy@basho.com>').
 -export([start/0, stop/0, new_request/2]).
 
+-type headers() :: webmachine_headers:t().
+-type response_body() :: binary()
```
Comment (seancribbs) [inline src/webmachine.erl]: Must it be `binary()`? I thought `iodata()` was allowed

File: src/wrq.erl [hunk 2]
```diff
@@ -204,65 +239,62 @@ merge_resp_headers(Hdrs, RD=#wm_reqdata{resp_headers=RespH}) ->
...
-            end;
-        https ->
-            case Port of
-                443 -> "";
-                _ -> ":" ++ erlang:integer_to_list(Port)
-            end;
-        _ -> ":" ++ erlang:integer_to_list(Port)
-    end.
+-spec port_string(scheme(), inet:port_number()) -> string().
+port_string(http, 80) -> "";
```
Comment (seancribbs) [inline src/wrq.erl:294]: 👏

**Review by seancribbs — CHANGES_REQUESTED** [review]
Other than the `binary()` / `iodata()` thing (needs investigation), this looks really good. Dialyzer likes it!

File: src/webmachine.erl (hunk 1 above)
Comment (joedevivo) [reply src/webmachine.erl]: You're right, I hit something else that expects `iodata()`

**Review by seancribbs — APPROVED** [review]
lgtm 💯

**Pull request merged**